packages/x-charts/src/context/CartesianProvider/computeValue.ts

```diff
@@ -144,10 +160,14 @@ export function computeValue(
     const scaleType = axis.scaleType ?? ('linear' as const);
 
     const extremums = [axis.min ?? minData, axis.max ?? maxData];
-    const tickNumber = getTickNumber({ ...axis, range, domain: extremums });
+    const tickNumber =
+      getTickNumber({ ...axis, range, domain: extremums }) / ((zoomRange[1] - zoomRange[0]) / 100);
+
+    const zoomedRange = zoomedScaleRange(range, zoomRange);
 
-    const scale = getScale(scaleType, extremums, range).nice(tickNumber);
-    const [minDomain, maxDomain] = scale.domain();
+    const scale = getScale(scaleType, extremums, zoomedRange);
+    // TODO: move nice to prop? Disable when there is zoom?
+    const [minDomain, maxDomain] = scale.nice(tickNumber).domain();
```

Comment:
About your comment: It's a good surprise from using the range manipulation: We do not impact the domain by zooming

And the `.nice` is relying on the domain, not the range.
https://github.com/d3/d3-scale/blob/main/src/linear.js

A solution could be to compute a `rawTickNumber = getTickNumber({ ...axis, range, domain: extremums })`

Use this `rawTickNumber` as an argument of the `nice()` function, and after compute. Such that it stays independent from the zooming.

`tickNumber = rawTickNumber / ((zoomRange[1] - zoomRange[0]) / 100)`